### ISSUE TITLE
Compiler warning

### ISSUE THREAD

**Issue by krlmlr** [rstats-db/RMariaDB#3]
<a href="https://github.com/jeroen"><img src="https://avatars0.githubusercontent.com/u/216319?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [jeroen](https://github.com/jeroen)**
_Tuesday Feb 24, 2015 at 20:12 GMT_
_Originally opened as https://github.com/rstats-db/RMySQL/issues/52_

----

@hadley I think this was introduced recently?

``` c
In file included from ./MyBinding.h:6:
./MyTypes.h:19:10: warning: enumeration values 'MYSQL_TYPE_TIMESTAMP2',
      'MYSQL_TYPE_DATETIME2', and 'MYSQL_TYPE_TIME2' not handled in switch
      [-Wswitch]
  switch(type) {
         ^
./MyTypes.h:61:1: warning: control may reach end of non-void function
      [-Wreturn-type]
}
^
2 warnings generated.
```

**Comment by krlmlr** [comment]
<a href="https://github.com/hadley"><img src="https://avatars0.githubusercontent.com/u/4196?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [hadley](https://github.com/hadley)**
_Tuesday Feb 24, 2015 at 20:43 GMT_

----

Oh I bet it's with your libmariadb update. I'll take a look

**Comment by krlmlr** [comment]
<a href="https://github.com/jeroen"><img src="https://avatars0.githubusercontent.com/u/216319?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [jeroen](https://github.com/jeroen)**
_Wednesday Feb 25, 2015 at 07:00 GMT_

----

This is actually on osx.

**Comment by krlmlr** [comment]
<a href="https://github.com/hadley"><img src="https://avatars0.githubusercontent.com/u/4196?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [hadley](https://github.com/hadley)**
_Wednesday Feb 25, 2015 at 13:35 GMT_

----

Hmmm, maybe you and I have different client versions? I don't seem to get this error locally.

I wonder what the best way of handling this is. I need to provide a default statement for that switch, but what should it do? It could either throw an error, or it might be possible to use character for unknown variable types, and hope MySQL coercion does something reasonable.

**Comment by krlmlr** [comment]
All compiler warnings fixed in 437fedfd.
